vllm/v1/attention/backends/short_conv_attn.py

```diff
@@ -79,14 +71,22 @@ def build(self,
             nums_dict, batch_ptr, token_chunk_offset_ptr = \
                 compute_causal_conv1d_metadata(query_start_loc_p)
 
+        elif (num_decodes > 0 and num_decodes <= self.decode_cudagraph_max_bs
+              and self.compilation_config.full_cuda_graph):
+            num_input_tokens = self.vllm_config.pad_for_cudagraph(num_decodes)
+            self.state_indices_tensor[:num_decodes].copy_(state_indices_tensor,
+                                                          non_blocking=True)
+            state_indices_tensor = self.state_indices_tensor[:num_input_tokens]
+            state_indices_tensor[num_decodes:] = PAD_SLOT_ID
+
         attn_metadata = ShortConvAttentionMetadata(
+            query_start_loc=query_start_loc,
```

Comment:
Another small optimization could be to move [the code](https://github.com/vllm-project/vllm/blob/main/vllm/v1/attention/backends/mamba2_attn.py#L184-L185) to compute `query_start_loc_p` into the metadata builder, and then store `query_start_loc_p` instead of `query_start_loc` in `ShortConvAttentionMetadata`. Doesn't necessarily need to be done as part of this PR though.